### PR TITLE
PWX-16095: Update spec to support CSI cloud drives

### PR DESCRIPTION
**Pull request by trierra**

### CODE REVIEW

**Review by piyush-nimbalkar — APPROVED** [review]
The tests are going to break, you will have to fix those. 
You will have to fix the test spec file(s) to reflect the latest permissions.

File: drivers/storage/portworx/component/portworx_basic.go [hunk 1]
```diff
@@ -262,8 +262,8 @@ func (c *portworxBasic) createClusterRole() error {
 				},
 				{
 					APIGroups: []string{"storage.k8s.io"},
-					Resources: []string{"storageclasses"},
-					Verbs:     []string{"get", "list"},
+					Resources: []string{"storageclasses", "csinodes", "volumeattachments"},
+					Verbs:     []string{"get", "list", "create", "delete", "update"},
```
Comment (harsh-px) [inline drivers/storage/portworx/component/portworx_basic.go:266]: this is giving more privileges to storageclasses than needed. See comment: https://github.com/portworx/px-installer/pull/958#discussion_r595588078

**Review by harsh-px — CHANGES_REQUESTED** [review]

File: drivers/storage/portworx/component/portworx_basic.go [hunk 2]
```diff
@@ -262,9 +262,14 @@ func (c *portworxBasic) createClusterRole() error {
 				},
 				{
 					APIGroups: []string{"storage.k8s.io"},
-					Resources: []string{"storageclasses"},
+					Resources: []string{"storageclasses", "csinodes", "volumeattachments"},
```
Comment (harsh-px) [inline drivers/storage/portworx/component/portworx_basic.go:265]: `volumeattachments` is redudant. Also present on line 270.

Comment (trierra) [reply drivers/storage/portworx/component/portworx_basic.go:265]: ah, didn't remove one, fixing

**Review by harsh-px — APPROVED** [review]

**Pull request merged**